### ISSUE TITLE
AArch64: Failure in testDDRExt_SharedClasses_openj9_0

### ISSUE THREAD

**Issue by knn-k** [eclipse/openj9#9003]
Failure link
https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_extended.functional_aarch64_linux_Personal/7/consoleText

testDDRExt_SharedClasses_openj9_0 in extended.functional fails with AArch64 as follows, failing in finding AOT data.
```
[2020-03-23T02:06:05.323Z]      [java] [AutoRun] [INFO] Starting testShrcAOTStatsExt(j9vm.test.ddrext.junit.TestSharedClassesExt)
[2020-03-23T02:06:05.638Z]      [java] [DDRExtTesterBase] [ERROR] '!shrc  aotstats' output does not contain success key : Ljava/lang/Object
[2020-03-23T02:06:05.638Z]      [java] [DDRExtTesterBase] [ERROR] '!shrc  aotstats' output :!j9sharedclassconfig 0x0000FFFFB01DF660
[2020-03-23T02:06:05.638Z]      [java] 
[2020-03-23T02:06:05.638Z]      [java] Meta data region to be used: 0x0000FFFF6E7FB634..0x0000FFFF6E80FFFC
[2020-03-23T02:06:05.638Z]      [java] No entry found in the cache
[2020-03-23T02:06:05.638Z]      [java] 
[2020-03-23T02:06:05.638Z]      [java] Cache contains 737 classes, 11 orphans, 3 classpaths, 0 URLs, 0 tokens
[2020-03-23T02:06:05.638Z]      [java] 0 AOT, 13 SCOPES, 5 BYTE data, 0 UNINDEXED DATA, 0 CHARARRAY, 0 stale
```

**Comment by knn-k** [comment]
The test passes with the following output when I run the test on a local machine:
```
     [java] [AutoRun] [INFO] Starting testShrcAOTStatsExt(j9vm.test.ddrext.junit.TestSharedClassesExt)
     [java] [DDRExtTesterBase] [INFO] Validation started for : '!shrc  aotstats'
     [java] [DDRExtTesterBase] [INFO] Validation passed for : '!shrc  aotstats'
     [java] [AutoRun] [INFO] Finished testShrcAOTStatsExt(j9vm.test.ddrext.junit.TestSharedClassesExt)
```

**Comment by knn-k** [comment]
`printStats` shows that no methods are AOTed by ` -Xscmx16m -Xshareclasses:name=ddrextjunitSCC,addtestjithints -Xjit:count=0 -Xaot:forceaot,count=5,disableAsyncCompilation` on the test server, as shown below:
```
     [java] Current statistics for cache "ddrextjunitSCC":
     (... skip ...)
     [java] cache layer                          = 0
     [java] cache size                           = 314572192
     [java] softmx bytes                         = 67108864
     [java] free bytes                           = 63976622
     [java] Reserved space for AOT bytes         = -1
     [java] Maximum space for AOT bytes          = -1
     [java] Reserved space for JIT data bytes    = -1
     [java] Maximum space for JIT data bytes     = -1
     [java] Metadata bytes                       = 47268
     [java] Metadata % used                      = 1%
     [java] Class debug area size                = 25100288
     [java] Class debug area used bytes          = 540666
     [java] Class debug area % used              = 2%
     [java]
     [java] ROMClass bytes                       = 2441768
     [java] AOT bytes                            = 400
     [java] JIT data bytes                       = 36484
     [java] Zip cache bytes                      = 0
     [java] Startup hint bytes                   = 120
     [java] Data bytes                           = 64928
     [java]
     [java] # ROMClasses                         = 741
     [java] # AOT Methods                        = 0
     [java] # Classpaths                         = 3
     [java] # URLs                               = 0
     [java] # Tokens                             = 0
     [java] # Zip caches                         = 0
     [java] # Startup hints                      = 1
     [java] # Stale classes                      = 0
     [java] % Stale classes                      = 0%
     [java]
     [java]
     [java] Cache is 4% soft full
     [java]
     [java] Cache is accessible to current user = true
```

**Comment by knn-k** [comment]
On the other hand, doing the same on a local machine generates the following output.  229 methods are AOTed.
```
     [java] Current statistics for cache "ddrextjunitSCC":
     (... skip ...)
     [java] cache layer                          = 0
     [java] cache size                           = 314572192
     [java] softmx bytes                         = 67108864
     [java] free bytes                           = 63509020
     [java] Reserved space for AOT bytes         = -1
     [java] Maximum space for AOT bytes          = -1
     [java] Reserved space for JIT data bytes    = -1
     [java] Maximum space for JIT data bytes     = -1
     [java] Metadata bytes                       = 61380
     [java] Metadata % used                      = 1%
     [java] Class debug area size                = 25161728
     [java] Class debug area used bytes          = 540068
     [java] Class debug area % used              = 2%
     [java]
     [java] ROMClass bytes                       = 2446752
     [java] AOT bytes                            = 487224
     [java] JIT data bytes                       = 35628
     [java] Zip cache bytes                      = 0
     [java] Startup hint bytes                   = 120
     [java] Data bytes                           = 28064
     [java]
     [java] # ROMClasses                         = 739
     [java] # AOT Methods                        = 229
     [java] # Classpaths                         = 3
     [java] # URLs                               = 0
     [java] # Tokens                             = 0
     [java] # Zip caches                         = 0
     [java] # Startup hints                      = 1
     [java] # Stale classes                      = 0
     [java] % Stale classes                      = 0%
     [java]
     [java]
     [java] Cache is 5% soft full
     [java]
     [java] Cache is accessible to current user = true
```

**Comment by knn-k** [comment]
You get the following 9 failures in this test when you use `IBM_JAVA_OPTIONS=-Xquickstart` to force AOT compilation.
```
     [java] [AutoRun] [ERROR] testShrcStatsExt(j9vm.test.ddrext.junit.TestSharedClassesExt)
     [java] [AutoRun] [ERROR] testShrcAllStatsExt(j9vm.test.ddrext.junit.TestSharedClassesExt)
     [java] [AutoRun] [ERROR] testShrcRcStatsExt(j9vm.test.ddrext.junit.TestSharedClassesExt)
     [java] [AutoRun] [ERROR] testShrcCpStatsExt(j9vm.test.ddrext.junit.TestSharedClassesExt)
     [java] [AutoRun] [ERROR] testShrcJITPStatsExt(j9vm.test.ddrext.junit.TestSharedClassesExt)
     [java] [AutoRun] [ERROR] testShrcFindJITPext(j9vm.test.ddrext.junit.TestSharedClassesExt)
     [java] [AutoRun] [ERROR] testShrcFindJITPPext(j9vm.test.ddrext.junit.TestSharedClassesExt)
     [java] [AutoRun] [ERROR] testShrcJITPForExt(j9vm.test.ddrext.junit.TestSharedClassesExt)
     [java] [AutoRun] [ERROR] testShrcJITHStatsExt(j9vm.test.ddrext.junit.TestSharedClassesExt)
```

**Comment by knn-k** [comment]
I can make this test pass by setting `IBM_JAVA_OPTIONS=-Xaot:forceAot`, either on cent7-aarch64-1 or in my development environment.

**Comment by knn-k** [comment]
The error message with `IBM_JAVA_OPTIONS=-Xquickstart` is:
> [DDRExtTesterBase] [ERROR] '!shrc  stats' output does not contain success key : JITPROFILE data length [1-9][0-9]* metadata [1-9][0-9]*

The output contains a line:
> JITPROFILE data length 0 metadata 0

Running with `IBM_JAVA_OPTIONS=-Xaot:forceAot`, the line looks like:
> JITPROFILE data length 10824 metadata 2976

**Comment by pshipton** [comment]
I notice in the PR build https://ci.eclipse.org/openj9/job/Pipeline_Build_Test_JDK11_aarch64_linux/72/ this test isn't listed as disabled, passing or failing.

**Comment by knn-k** [comment]
@pshipton This test was disabled by PR #8569, and PR #8764 (not merged) is for enabling it again.

**Comment by pshipton** [comment]
#8764 enables testDDRExt_JITExt_openj9, not estDDRExt_SharedClasses_openj9.

**Comment by pshipton** [comment]
What is the plan for this test for the 0.20.0 release? Will it remain excluded or is a workaround planned?

**Comment by knn-k** [comment]
Yesterday I changed #8764 to leave `testDDRExt_SharedClasses_openj9` excluded.
I don't want to introduce new AOT heuristics in the JIT for platforms without recompilation at the final moment before release.  I am going to enable this test when I finish implementing recompilation for AArch64.

I have confirmed that the functionality of DDR works as expected by enabling AOT with a command-line option.

FYI. @0xdaryl
